consensus/src/transactions/free.rs

```diff
@@ -1,9 +1,40 @@
+use std::sync::Mutex as StdMutex;
+
 use monero_serai::{
     ringct::{bulletproofs::Bulletproof, RctType},
     transaction::{Input, Transaction},
 };
 
-use cuprate_consensus_rules::transactions::TransactionError;
+use cuprate_consensus_rules::{transactions::TransactionError, ConsensusError};
+use cuprate_types::{CachedVerificationState, TransactionVerificationData, TxVersion};
+
+/// Creates a new [`TransactionVerificationData`] from a [`Transaction`].
+///
+/// # Errors
+///
+/// This function will return [`Err`] if the transaction is malformed, although returning [`Ok`] does
+/// not necessarily mean the tx is correctly formed.
+pub fn new_tx_verification_data(
+    tx: Transaction,
+) -> Result<TransactionVerificationData, ConsensusError> {
```

Comment:
The error is defined in this crate, I did try defining a new error type and mapping between them but it is very messy,

I don't really want to move the error definition.

I think this is more clear as a free function rather than a `TryFrom`, plus both types are defined outside this crate